DOC: Clarify a point in basic indexing user guide

djramones opened the pull request:
Minor doc enhancement to address a confusing statement I encountered in the user guide.

Fixes #24769.

djramones committed:
DOC: Clarify a point in basic indexing user guide

rossbar commented on doc/source/user/basics.indexing.rst:
```diff
@@ -194,8 +194,16 @@ concepts to remember include:
-  elements *i*, *i+k*, ..., *i + (m - 1) k < j*,
+  stacking, along the *p*-th axis, the sub-arrays returned by integer
+  indexing of elements *i*, *i+k*, ..., *i + (m - 1) k < j*. This means
+  that, for example, with the array::
+
+      >>> x = np.arange(24).reshape(2, 4, 3)
+
+  The following are equivalent::
+
+      >>> x[:, 1:3, :]
+      >>> np.stack([x[:, i, :] for i in range(1, 3)], axis=1)
```
The main issue here is that the resulting arrays have the same values, but the operations are not equivalent as the first returns a view whereas the `stack` returns a new array.

I understand the desire to demonstrate the indexing with an example, but I don't think this is a good example to add in the context of basic indexing.

djramones replied:
Ah, you're right, the return values are not truly equivalent.

I can add a further explanation of that caveat (and fix the failing doctests), or are you saying it's too complicated for this part of the doc? Should I remove the example altogether?

rossbar replied:
I don't have a strong preference, but my vote would be to drop the example to avoid confusion. The wording update (concatenating subarrays -> stacking along axis) seems fine, though YMMV whether it improves clarity

djramones committed:
DOC: Remove proposed basic indexing example

The example could be more confusing than helpful in this context.
Besides, the claim that the examples are equivalent is not accurate (one
returns a view, the other returns a new array).

rossbar reviewed (approved):
Thanks for the updates @djramones , LGTM!
